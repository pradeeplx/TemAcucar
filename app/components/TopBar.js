import React from 'react'
import { Platform, View, Text, TouchableOpacity, StyleSheet } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"
import LogoIcon from "./LogoIcon"
import Icon from "./Icon"

export default TopBar = ({ onMenuOpen }) => (
  <View style={{
    marginTop: (Platform.OS == 'ios' ? 20 : 0),
    height: 40 * fontFactor(),
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderColor: Colors.pink,
    borderBottomWidth: 1,
  }}>
    <View style={{
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
    }}>
      <Sentence style={[{
        fontSize: 14 * fontFactor(),
      }, (Platform.OS === 'ios' ? { fontFamily: 'Avenir-Black' } : { fontWeight: 'bold' })]}>
        Tem Açúcar?
      </Sentence>
    </View>
    <TouchableOpacity onPress={onMenuOpen} style={{
      paddingHorizontal: 10,
      position: 'absolute',
      top: 0,
      left: 0,
      bottom: 0,
      justifyContent: 'center',
    }}>
      <Icon name="ios-menu" style={{
        color: Colors.brown,
        fontSize: 28 * fontFactor(),
      }} />
    </TouchableOpacity>
  </View>
)
