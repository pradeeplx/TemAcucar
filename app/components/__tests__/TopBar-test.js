import 'react-native'
import React from 'react'
import TopBar from '../TopBar'
import renderer from 'react-test-renderer'
import { identity } from 'lodash'

it('renders correctly', () => {
  const tree = renderer.create(
    <TopBar style={{top: 0}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
