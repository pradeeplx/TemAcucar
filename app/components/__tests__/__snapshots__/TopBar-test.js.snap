exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "#FF7F85",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "marginTop": 20,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "fontSize": 21,
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Tem Açúcar?
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "opacity": 1,
        "paddingHorizontal": 10,
        "position": "absolute",
        "top": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "color": "#654C50",
            "fontSize": 42,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
